### PR TITLE
FE - 1.5: Firebase Auth: Email/Password

### PR DESCRIPTION
**Pull request by lxyhan**
Bug Fix: Resolved an issue in Google Authentication with TypeScript integration.
Sign-In/Sign-Up Form: Built a dynamic form supporting both email/password and Google sign-ins, with a toggle for sign-up.
Error Handling: Implemented detailed error messages for form validation and Firebase errors (e.g., invalid credentials, weak passwords, etc.).
Firebase Email Auth: Fully set up Firebase Email Authentication for user registration and login.

### CODE REVIEW

**Comment by lxyhan** [comment]
> I'm wondering if it would be useful to ensure that the inputs are valid before clicking `Sign Up with Email`. For example, an error message could be displayed while the user types in their email if it doesn't match a valid format, or password strength hints could be displayed dynamically instead of when you submit the form.

Hi Stephanie! Thanks for the suggestion, and I agree that input checking would be useful. I'll make another commit as soon as possible and I hope you can review the PR after I'm done. Thanks again!

**Comment by stephanieyflu** [comment]
LGTM! Looks like there's a conflict though

**Review by stephanieyflu — APPROVED** [review]
LGTM! Looks like there's a conflict though

File: frontend/src/pages/SignIn/SignIn.tsx [hunk 1]
```diff
@@ -26,8 +26,14 @@ const Login: React.FC = () => {
 
     // Handle Google sign-in
     const handleGoogleSignIn = () => {
+        localStorage.setItem("AddedToGame", "false"); // Reset added to game status
```
Comment (Ron-Madan) [inline frontend/src/pages/SignIn/SignIn.tsx:29]: nit: "addedToGame"

**Review by Ron-Madan — APPROVED** [review]
LGTM!

**Review by Ron-Madan — APPROVED** [review]
LGTM!

**Pull request merged**